src/app/frontend/replicasetdetail/replicasetdetail.html

```diff
@@ -138,8 +138,8 @@
           </md-input-container>
           <md-input-container class="kd-replicasetdetail-option-picker">
             <label>Source</label>
-            <!-- TODO(maciaszczykm): Handle value change. -->
-            <md-select ng-model="ctrl.eventSource" required>
+            <md-select ng-model="ctrl.eventSource"
+                       ng-change="ctrl.handleEventFiltering()" required>
```

Comment:
Why don't you create two `@exported` methods and use them here in this template. E.g., `handleFilterEventsBySource` and `handleFilterEventsByType`. I dont see a reason why this method does both.